diesel/src/query_dsl/mod.rs

```diff
@@ -603,6 +618,14 @@ pub trait QueryDsl: Sized {
     ///
     /// If there is already a `WHERE` clause, the result will be `old AND new`.
     ///
+    /// Note that in order to use this method with a method like [`.inner_join`] or [`.select`], you will need to use them before calling
+    /// [`.filter`] (See examples below). This is because you can only access columns from tables
+    /// that appear in your query before the call to [`.filter`].
+    ///
+    /// [`.inner_join`]: QueryDsl::inner_join()
+    /// [`.select`]: QueryDsl::select()
+    /// [`.filter`]: QueryDsl::filter()
+    ///
```

Comment:
Similary to the `select` comment, this comment should not mention `select` as the order of these two functions does not matter. I would also suggest to remove the `See examples below` sentence here, as the example below does not show that.

```suggestion
    /// Note that in order to use this method with columns from different tables, you need to call
    //  [`.inner_join`] or [`.left_join`] beforehand.
    /// This is because you can only access columns from tables
    /// that appear in your query before the call to [`.filter`].
    ///
    /// [`.inner_join`]: QueryDsl::inner_join()
    /// [`.left_join`]: QueryDsl::left_join()
    /// [`.filter`]: QueryDsl::filter()
    ///
```